src/main/java/com/foo/gosucatcher/domain/expert/presentation/ExpertController.java

```diff
@@ -49,17 +64,66 @@ public ResponseEntity<ExpertsResponse> findAll() {
 		return ResponseEntity.ok(experts);
 	}
 
-	@PatchMapping("/{id}")
-	public ResponseEntity<Long> update(@PathVariable Long id, @Validated @RequestBody ExpertUpdateRequest request) {
-		Long expertId = expertService.update(id, request);
+	@PatchMapping("/{expertId}")
+	public ResponseEntity<Long> update(@PathVariable Long expertId,
+		@Validated @RequestBody ExpertUpdateRequest request) {
+		Long updatedExpertId = expertService.update(expertId, request);
+
+		return ResponseEntity.ok(updatedExpertId);
+	}
+
+	@DeleteMapping("/{expertId}")
+	public ResponseEntity<Void> delete(@PathVariable Long expertId) {
+		expertService.delete(expertId);
+
+		return ResponseEntity.ok(null);
+	}
+
+	@PostMapping("/{expertId}/images")
+	public ResponseEntity<ImageUploadResponse> uploadImage(@PathVariable Long expertId, MultipartFile file) throws
+		IllegalStateException,
+		IOException {
+		ImageUploadRequest request = new ImageUploadRequest(expertId, file);
+		String uploadedFilename = imageService.store(request);
 
-		return ResponseEntity.ok(expertId);
+		ImageUploadResponse response = new ImageUploadResponse(expertId, uploadedFilename);
+		return ResponseEntity.status(HttpStatus.CREATED).body(response);
 	}
 
-	@DeleteMapping("/{id}")
-	public ResponseEntity<Void> delete(@PathVariable Long id) {
-		expertService.delete(id);
+	@GetMapping("/{expertId}/images/{filename}")
+	public ResponseEntity<Resource> getImage(@PathVariable Long expertId,
+		@PathVariable String filename) {
+
+		Resource file = imageService.loadAsResource(expertId, filename);
+		return ResponseEntity.ok().header(HttpHeaders.CONTENT_DISPOSITION,
+			"attachment; filename=\"" + file.getFilename() + "\"").body(file);
+	}
 
+	@DeleteMapping("/{expertId}/images/{filename}")
+	public ResponseEntity<String> deleteImage(@PathVariable Long expertId,
+		@PathVariable String filename) {
+		imageService.delete(expertId, filename);
 		return ResponseEntity.ok(null);
 	}
-}
\ No newline at end of file
+
+	@GetMapping("{expertId}/images")
+	public ResponseEntity<List<ImageResponse>> getImages(@PathVariable Long expertId) {
+		List<ImageResponse> fileInfos = imageService.loadAll(expertId)
+			.map(path -> {
+				String filename = path.getFileName().toString();
+				String url = MvcUriComponentsBuilder.fromMethodName(ExpertController.class,
+					"getImage", expertId, filename).build().toString();
+				Long size;
+				try {
+					size = Files.size(path);
+				} catch (IOException e) {
+					size = 0L;
+				}
+				return new ImageResponse(filename, url, size);
+			})
+			.collect(Collectors.toList());
+
+		return ResponseEntity.status(HttpStatus.OK).body(fileInfos);
```

Comment:
취향의 문제긴 하지만 이런 방법도 가능해요~

```suggestion
		return ResponseEntity.ok(fileInfos);
```